src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -167,13 +171,10 @@ public void updatePerson(ReadOnlyPerson target, ReadOnlyPerson editedReadOnlyPer
     /**
      *Adds an insurance to the address book.
      */
-    public void addInsurance(ReadOnlyInsurance i) {
+    public void addInsurance(ReadOnlyInsurance i)
+            throws DuplicateInsuranceException, DuplicateInsuranceContractNameException {
         LifeInsurance lifeInsurance = new LifeInsurance(i);
-        try {
-            lifeInsuranceMap.put(lifeInsurance);
-        } catch (DuplicateInsuranceException e) {
-            assert false : "AddressBooks should not have duplicate insurances";
-        }
+        lifeInsuranceMap.put(lifeInsurance.getId(), lifeInsurance);
```

Comment:
Here instead of generating a `UUID` object as the key, I retrieved and passed in the one from `lifeInsurance` to ensure that no mismatching occurs.